### PR TITLE
Dashboard: Use same min refresh interval fallback in the API as in provisioning

### LINKED ISSUE #25717: Minimum refresh interval blocks dashboard from importing using api
**What happened**:
settings GF_DASHBOARDS_MIN_REFRESH_INTERVAL=30s blocks importing dashboards using api
`POST /api/dashboards/db`

**What you expected to happen**:
setting minimum refresh interval to 30s

**How to reproduce it (as minimally and precisely as possible)**:
Deploy Grafana 7.0.3 with GF_DASHBOARDS_MIN_REFRESH_INTERVAL=30s, try to import a Dashboard with less interval

`docker run -d -p 3000:3000 -e GF_DASHBOARDS_MIN_REFRESH_INTERVAL=30s  grafana/grafana:7.0.3`
dashboard.json:
```
{
  "dashboard": {
    "id": null,
    "uid": null,
    "title": "Production Overview",
    "tags": [ "templated" ],
    "timezone": "browser",
    "schemaVersion": 16,
    "version": 0,
    "refresh": "25s"
  },
  "folderId": 0,
  "overwrite": false
}
```
```
curl -i -X POST -H "Content-Type: application/json" http://admin:admin@localhost:3000/api/dashboards/db --data-binary "@dashboard.json"
HTTP/1.1 400 Bad Request
Cache-Control: no-cache
Content-Type: application/json
Expires: -1
Pragma: no-cache
X-Frame-Options: deny
Date: Fri, 19 Jun 2020 14:43:06 GMT
Content-Length: 51

{"message":"Dashboard refresh interval is too low"}
```

**Anything else we need to know?**:
Logs: 
```t=2020-06-19T13:35:38+0000 lvl=info msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=POST path=/api/dashboards/db status=400 remote_addr=127.0.0.1 time_ms=37 size=51 referer=```

Response from Grafana Api:
```{"err":"Unexpected status code returned from Grafana API (got: 400, expected: 200, msg:{\"message\":\"Dashboard refresh interval is too low\"})"}```

related to this one: https://github.com/grafana/grafana/issues/23099
**Environment**:
- Grafana version: 7.0.3
- OS Grafana is installed on: K8s 1.15
- User OS & Browser: Chrome

### PR DESCRIPTION
**Pull request by Kirchen99**
…ndle min refresh interval when creating/updating dashboard via API

**What this PR does / why we need it**:
If refresh interval is lower than minimum refresh interval
when creating/updating dashboard via API, use the minimum refresh interval
instead of returning error.

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/25717

**Special notes for your reviewer**:

The API should mirror the behavior of importing and provisioning.
@sakjur

### CODE REVIEW

**Comment by papagian** [comment]
Thank you for reporting this and for contribution and sorry for delayed response (I had completely missed it).
The modification looks great but you have to synchronise your branch with the latest master and resolving the conflict before being able to merge it.
Also, I would suggest updating [the API docs](https://github.com/grafana/grafana/blob/master/docs/sources/http_api/dashboard.md#create--update-dashboard) as well by adding `refresh` field in the example and in the JSON Body schema something like:

`- **refresh** - Set the dashboard refresh interval. If this is lower than [the minimum refresh interval]({{< relref "../administration/configuration.md#min_refresh_interval">}}) then it will be ignored and the minimum refresh interval will be enforced.
`

**Comment by Kirchen99** [comment]
> Thank you for reporting this and for contribution and sorry for delayed response (I had completely missed it).
> The modification looks great but you have to synchronise your branch with the latest master and resolving the conflict before being able to merge it.
> Also, I would suggest updating [the API docs](https://github.com/grafana/grafana/blob/master/docs/sources/http_api/dashboard.md#create--update-dashboard) as well by adding `refresh` field in the example and in the JSON Body schema something like:
> 
> `- **refresh** - Set the dashboard refresh interval. If this is lower than [the minimum refresh interval]({{< relref "../administration/configuration.md#min_refresh_interval">}}) then it will be ignored and the minimum refresh interval will be enforced. `

Thank you for the feedback. I have synchronised the fork and updated the API docs.

**Review by sakjur — APPROVED** [review]
✅ Looks great, thank you for contributing to Grafana!

**Review by papagian — APPROVED** [review]

**Comment by papagian** [comment]
@oddlittlebird can you have a look in the docs modification and if it seems fine to you we can merge it!

File: docs/sources/http_api/dashboard.md [hunk 1]
```diff
@@ -59,6 +60,7 @@ JSON Body schema:
 - **folderId** – The id of the folder to save the dashboard in.
 - **overwrite** – Set to true if you want to overwrite existing dashboard with newer version, same dashboard title in folder or same dashboard uid.
 - **message** - Set a commit message for the version history.
+- **refresh** - Set the dashboard refresh interval. If this is lower than [the minimum refresh interval]({{< relref "../administration/configuration.md#min_refresh_interval">}}) then Grafana will ignore it and will enforce the minimum refresh interval.
```
Comment (oddlittlebird) [inline docs/sources/http_api/dashboard.md:63]:
```suggestion
- **refresh** - Set the dashboard refresh interval. If this is lower than [the minimum refresh interval]({{< relref "../administration/configuration.md#min_refresh_interval">}}), then Grafana will ignore it and will enforce the minimum refresh interval.
```

**Review by oddlittlebird — APPROVED** [review]
LGTM! Thank you for this contribution!

**Pull request merged**